File: 4.counting_elements/lesson_4_frog_river_one.py
def solution(X, A):
    if(len(A) == 1):
        if(A[0] == 1):
            return 0
        else:
            return -1

    first_occurance = [-1] * X
    for time, position in enumerate(A):
        if(first_occurance[position - 1] == -1):
            first_occurance[position - 1] = time
    maxi = 0
    for occurance in first_occurance:
        if(occurance == -1):
            return -1
        if(occurance > maxi):
            maxi = occurance
    return maxi

File: 4.counting_elements/lesson_4_frog_river_one.js
function solution(X, A) {
    let map = new Map()
    for(let i = 1; i <= X; i++) {
        map.set(i, 1)
    }

    for(let i = 0; i < A.length; i++) {
        if(map.has(A[i])) {
            map.delete(A[i])
        }
        if(map.size == 0) {
            return i
        }
    }
    return -1
}